From 6f1550a522f0589d39c1765bc3b524e4803ca591 Mon Sep 17 00:00:00 2001
From: Vikas Katariya <vikas.katariya@arm.com>
Date: Fri, 10 Feb 2023 14:23:50 +0000
Subject: [PATCH] cassini-bsp: Enable parse service to use TS

Update the default `config.toml` to use trusted service provider for
parsec-service.

Upstream-Status: Inappropriate [Can only be used when platform supports
trusted services]

Signed-off-by: Vikas Katariya <vikas.katariya@arm.com>
Signed-off-by: Gowtham Suresh Kumar <gowtham.sureshkumar@arm.com>
---
 config.toml | 20 ++++++++++----------
 1 file changed, 10 insertions(+), 10 deletions(-)

diff --git a/config.toml b/config.toml
index 9532f32..d65bdbe 100644
--- a/config.toml
+++ b/config.toml
@@ -21,7 +21,7 @@
 #log_level = "warn"
 
 # Control whether log entries contain a timestamp.
-#log_timestamp = false
+log_timestamp = true
 
 # Decide how large (in bytes) request bodies can be before they get rejected automatically.
 # Defaults to 1MB.
@@ -30,7 +30,7 @@
 # Decide whether detailed information about errors occuring should be included in log messages.
 # WARNING: the details might include sensitive information about the keys used by Parsec clients,
 # such as key names or policies
-#log_error_details = false
+log_error_details = true
 
 # Decide how large (in bytes) buffers inside responses from this provider can be. Requests that ask
 # for buffers larger than this threshold will be rejected. Defaults to 1MB.
@@ -48,13 +48,13 @@ listener_type = "DomainSocket"
 
 # (Required) Timeout of the read and write operations on the IPC channel. After the
 # timeout expires, the connection is dropped.
-timeout = 200 # in milliseconds
+timeout = 3600000 # in milliseconds
 
 # Specify the Unix Domain Socket path. The path is fixed and should always be the default one for
 # clients to connect. However, it is useful to change it for tests.
 # WARNING: If a file already exists at that path, the service will remove it before creating the
 # socket file.
-#socket_path = "/run/parsec/parsec.sock"
+socket_path = "/run/parsec/parsec.sock"
 
 # (Required) Authenticator configuration.
 # WARNING: the authenticator MUST NOT be changed if there are existing keys stored in Parsec.
@@ -126,10 +126,10 @@ manager_type = "SQLite"
 # ⚠ WARNING: Changing provider name after use will lead to loss of existing keys.
 # ⚠
 # (Optional) The name of the provider
-name = "mbed-crypto-provider"
+#name = "mbed-crypto-provider"
 
 # (Required) Type of provider.
-provider_type = "MbedCrypto"
+#provider_type = "MbedCrypto"
 
 # (Required) Name of key info manager that will support this provider.
 # NOTE: The key info manager only holds mappings between Parsec key name and Mbed Crypto ID, along
@@ -137,7 +137,7 @@ provider_type = "MbedCrypto"
 # Crypto library by default within the working directory of the service, NOT in the same location
 # as the mappings mentioned previously. If you want the keys to be persisted across reboots, ensure
 # that the working directory is not temporary.
-key_info_manager = "sqlite-manager"
+#key_info_manager = "sqlite-manager"
 
 # Example of a PKCS 11 provider configuration
 #[[provider]]
@@ -287,9 +287,9 @@ key_info_manager = "sqlite-manager"
 # ⚠ WARNING: Changing provider name after use will lead to loss of existing keys.
 # ⚠
 # (Optional) The name of the provider
-#name = "trusted-service-provider"
+name = "trusted-service-provider"
 # (Required) Type of provider.
-#provider_type = "TrustedService"
+provider_type = "TrustedService"
 
 # (Required) Name of key info manager that will support this provider.
-#key_info_manager = "sqlite-manager"
+key_info_manager = "sqlite-manager"
